app/assets/javascripts/discourse/app/initializers/subscribe-user-changes.js

```diff
@@ -7,8 +7,12 @@ export default {
 
     if (user) {
       const bus = container.lookup("message-bus:main");
+      const appEvents = container.lookup("service:app-events");
       bus.subscribe("/user", (data) => {
         user.setProperties(data);
+        Object.entries(data).forEach(([key, value]) =>
+          appEvents.trigger(key, value)
```

Comment:
This seems odd to me - you are going to trigger one event per key that changed? And the name of the event is the key? So for example if username changed the event would be called `username`? 

I would prefer one event for all changed attributes, and a namespaced name like `user:changed`

Reply:
Yes we already discussed about that here: https://github.com/discourse/discourse/pull/14501#discussion_r724046347
As there are only two events triggered here (as of today), using their keys allows to just send `pending_posts_count` and `drafts_count` (if I’m not mistaken for the latter). It’s simpler to just listen to the right event instead of having to listen to a more global one and then check if the key we want is in there.
I still could prefix the events with `counter:` as I suggested previously if you prefer?

Reply:
I'd totally forgotten! But yes, `count-updated:` or similar is necessary here. We can't have events without namespaces.